
import os
import sys
import numpy as np
import pandas as pd
import joblib
from src.CreditCardSegmentation.logger_file.logger_obj import logger
from src.CreditCardSegmentation.Exception.custom_exception import CustomException
from sklearn.model_selection import train_test_split
from src.CreditCardSegmentation.entity.config_entity import DataTransformationConfig





class DataTransformation:
    def __init__(self, config : DataTransformationConfig):
        
        self.config = config




    def data_split(self):
        try:

            logger.info(f'-----------Entered data_split method---------------')
        
            data = pd.read_csv(self.config.local_data_file)

            data = data[["BALANCE", 'PURCHASES', 'CREDIT_LIMIT']]

            train, test = train_test_split(data, test_size=0.2, random_state=42)

            train.to_csv(os.path.join(self.config.train_path, 'train.csv'), index = False, header = True)
        
            test.to_csv(os.path.join(self.config.test_path, 'test.csv'), index = False, header = True)

            logger.info(f'----------------saved train test data in csv format------------------')

            logger.info(f'------------The shape of the train data is {train.shape}')

            logger.info(f'--------------The shape of the test data is {test.shape}')

            logger.info(f'------------completed data splitting----------------------')

        except Exception as e:
            raise CustomException(e, sys)